download-basket/src/main/java/downloadbasket/helpers/OGCServices.java

```diff
@@ -76,60 +87,68 @@ public static String getFilter(JSONObject downloadDetails, Boolean writeParam)
 	 * @throws JSONException
 	 */
 	private static String getBbox(JSONObject bbox) throws JSONException {
-		return bbox.getString(PARAM_BBOX_LEFT) + "," + bbox.getString(PARAM_BBOX_BOTTOM) + ","
-				+ bbox.getString(PARAM_BBOX_RIGHT) + "," + bbox.getString(PARAM_BBOX_TOP);
+		return bbox.getString(KEY_BBOX_LEFT) + "," + bbox.getString(KEY_BBOX_BOTTOM) + ","
+				+ bbox.getString(KEY_BBOX_RIGHT) + "," + bbox.getString(KEY_BBOX_TOP);
 	}
 
 	/**
-	 * Get WFS Query layer WFS request by using GeoServer Cross-layer filtering plugin.
+	 * Get WFS Query layer WFS request by using GeoServer Cross-layer filtering
+	 * plugin.
 	 * 
 	 * @param download
 	 *            download details
+	 * @param geometryColumnName
+	 * 			  name of the geometry column in a downloadable layer, configured in layer attributes
+	 * @param cropGeomColumn
+	 * 			  name of the geometry column in a cropping layer, configured in layer attributes
+	 * @param uniqueColumn
+	 * 			  name of a column in a cropping layer, that contains an unique value such as an id. Configured in layer attributes
+	 *            
 	 */
-	
-	public static String getPluginFilter(JSONObject download)
-			throws JSONException {
-		JSONArray identifiers = new JSONArray(download.getString(PARAM_IDENTIFIERS));
+
+	public static String getPluginFilter(JSONObject download, OskariLayerService mapLayerService) throws JSONException {
+		JSONArray identifiers = new JSONArray(download.getString(KEY_IDENTIFIERS));
 		String xml = "";
 		String croppingNameSpace = PropertyUtil.get("oskari.wfs.cropping.namespace");
 
 		try {
 			ByteArrayOutputStream baos = new ByteArrayOutputStream();
 
+			OskariLayer oskariLayer = mapLayerService.find(download.getString(KEY_LAYER_ID));
```

Comment:
Why not just pass the layer as param if you don't need the service for anything else? It would make this function so much easier to test.